Sources/SwiftFormat/Core/WithAttributesSyntax+Convenience.swift

```diff
@@ -0,0 +1,28 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+extension WithAttributesSyntax {
+    /// Indicates whether the node has attribute with the given `name`.
+    ///
+    /// - Parameter name: The name of the attribute to lookup.
+    /// - Returns: True if the node has an attribute with the given `name`, otherwise false.
+    func hasAttribute(_ name: String) -> Bool {
+        attributes.contains { attribute in
+            let attributeName = attribute.as(AttributeSyntax.self)?.attributeName
+            return attributeName?.as(IdentifierTypeSyntax.self)?.name.text == name
+            // support @Module.Attribute syntax as well
+            || attributeName?.as(MemberTypeSyntax.self)?.name.text == name
```

Comment:
Since you're not checking the module name here, this would match anything, right? Like `@Foo.Test`, for example.

It might be better to make the function signature `hasAttribute(_ name: String, inModule: String)` and have the caller always provide it, but it would only be checked if it was a `MemberTypeSyntax`. So `hasAttribute("Test", inModule: "Testing")` would still match either `@Test` or `@Testing.Test`, but not `@Foo.Test`.

Reply:
What you think if we will have 2 functions, one will be the current one  and another function like you suggested?

Reply:
I think it's better for simplicity to have one function that does both checks. We can't _definitively_ know if `@Test` refers to `@Testing.Test` without the file being type-checked, which we can't do in swift-format. So we're making an assumption that `@Test` by itself refers to the one in `Testing` and nothing else, because of how common we expect that pattern to be. If we're handling that as the common case, then there's no reason to not also handle `@Testing.Test` with the same query—each of the individual rules shouldn't have to make two calls to find the attribute it's looking for.